### PR TITLE
Handle loading screens

### PR DESCRIPTION
**Pull request by kozaktar**
-Fixed bug where onboarding page would briefly appear when logging in. Now it displays loading screen instead.
-Added withSpinner component to make it easy to add lading page to any components.
Usage: (adding spinner to Modal component)

const ModalWithSpinner=WithSpinner(Modal);
 <ModalWithSpinner isloading={loadingState}/>
 
If loadingState is set to true loading indicator will be displayed instead of the component. If loadingState is set to false the component will be displayed as normal.

### CODE REVIEW

File: client/src/components/WithSpinner.js
Comment (dproc96) [inline client/src/components/WithSpinner.js:2]: Can you run prettier on this file?

File: client/src/pages/OnBoarding.js
Comment (dproc96) [inline client/src/pages/OnBoarding.js]: Can you run prettier on this file as well?

**Review by dproc96 — APPROVED** [review]
Pending some code formatting, this is good to go

**Comment by dproc96** [comment]
Actually hold off on merging this until Matt's changes have been merged in

File: client/src/components/LoadingScrean.js
Comment (mattchx) [inline client/src/components/LoadingScrean.js]: Good Styling!

File: server/routes/teamEventAPI.js
Comment (mattchx) [inline server/routes/teamEventAPI.js]: Did you add this in by accident ?

**Review by mattchx — CHANGES_REQUESTED** [review]
I'd suggest removing teamEventAPI.js

File: server/routes/teamEventAPI.js
Comment (kozaktar) [reply server/routes/teamEventAPI.js]: I don't even know how this got here. I don't remember changing any backend files....
Maybe I merged dev into this branch before creating the pull request?

**Pull request merged**
